gui/default/syncthing/core/syncthingController.js

```diff
@@ -3026,6 +3026,55 @@ angular.module('syncthing.core')
                     address.indexOf('unix://') == 0 ||
                     address.indexOf('unixs://') == 0);
         };
+
+        $scope.copyToClipboard = function (event, text) {
+            console.log(text);
+            var success = $translate.instant("Copied to clipboard!");
+            var failure = $translate.instant("Copy to clipboard failed!");
+
+            if (navigator.clipboard && navigator.clipboard.writeText) {
+                // Default for modern browsers on localhost or HTTPS.
+                navigator.clipboard.writeText(text);
+                return alert(success);
+            } else if (window.clipboardData && window.clipboardData.setData) {
+                // Fallback for Internet Explorer.
+                window.clipboardData.setData('Text', text);
+                return alert(success);
+            } else if (document.queryCommandSupported) {
+                // Fallback for modern browsers on HTTP and non-IE old browsers.
+                // Check for document.queryCommandSupported("copy") support is
+                // omitted on purpose, as old Chrome versions reported "false"
+                // despite supporting the feature.
+                var textarea = document.createElement("textarea");
+                event.currentTarget.appendChild(textarea);
+                textarea.style.position = "fixed";
+                textarea.style.opacity = "0";
+                textarea.textContent = text;
+                textarea.select();
+                try {
+                    document.execCommand("copy");
+                    return alert(success);
+                }
+                catch (ex) {
+                    return alert(failure);
+                }
+                finally {
+                    event.currentTarget.removeChild(textarea);
+                }
+            }
+            return alert(failure);
+        };
+
+        $scope.shareDeviceIdBy = function(channel) {
+            var body = $translate.instant("You have been invited to connect with {%name%} using Syncthing!", {name: $scope.currentDevice.name}) + "\r\n\r\n" + $translate.instant("Accept the invitation by adding a new device using the following ID.") + $translate.instant("If unsure what to do, continue reading below.") + "\r\n\r\n" + $scope.currentDevice.id + "\r\n\r\n" + $translate.instant("Syncthing is a continuous file synchronization program.") + $translate.instant("It synchronizes files between two or more computers in real time, safely protected from prying eyes.") + $translate.instant("Your data is your data alone and you deserve to choose where it is stored, whether it is shared with some third party, and how it's transmitted over the internet.") + "\r\n\r\n" + $translate.instant("For more information about Syncthing please visit https://syncthing.net.");
```

Comment:
I was worried about making this a single, very long string, as then even a single change will require re-translating everything. I will have a look whether it could be reduced to a few larger, separate paragraphs though.

Reply:
I've modified the code, so that strings like the title, invitation, and footer are still separated, but the long description text is kept as a single entity, which should prevent situations, where we end up with English sentences right in between translated ones.

I don't really believe we should lump the whole email/sms text together though, as this will make it basically impossible to add any fixes later on (e.g. in a possibility that the "learn more" URL changes, etc.).

Reply:
Looks better now, but please split once more on the first body line break / paragraph.